From e43d25f52493925ac1284f5fb2e780f0ceaf8b5d Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 19 Jun 2012 17:56:36 +0800
Subject: [PATCH 293/635] ENGR00214149 Camera preview is green for ov5642 and
 ov5640

Add new mode setting for XGA as driver update

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/V4l2CsiDevice.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 0f8da8a..f48f718 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -375,6 +375,11 @@ namespace android{
                 //pTimeFrame->numerator = 1;
                 //pTimeFrame->denominator = 30;
             }
+            else if (capturewidth == 1024 && captureheight == 768) {
+                capturemode = 8;       /* XGA mode */
+                //pTimeFrame->numerator = 1;
+                //pTimeFrame->denominator = 30;
+            }
             else{
                 CAMERA_HAL_ERR("The camera mode is not supported!!!!");
                 return CAPTURE_DEVICE_ERR_BAD_PARAM;
-- 
1.8.0

